src/history.js

```diff
@@ -1,28 +1,36 @@
-import Action from './action'
-
 /**
  * The central tree data structure that is used to calculate state for
  * a Microcosm. Each node in the tree represents an action. Branches
  * are changes over time.
  */
-export default function History (limit=0) {
+
+import Action from './action'
+
+export default function History (limit) {
+  this.ids = 0
+  this.size = 0
+  this.limit = Math.max(1, limit || 1)
   this.repos = []
-  this.limit = limit
+
+  this.append('_root', 'resolve')
 }
```

Comment:
History now _always_ has a root node. At the start, we append a completed placeholder.